### ISSUE TITLE
Only the hello_world_test test runs when configured with PIKA_WITH_TESTS PIKA_WITH_TESTS_UNIT PIKA_WITH_TESTS_REGRESSIONS

### ISSUE THREAD

**Issue by yurivict** [pika-org/pika#853]
## Expected Behavior

PIKA_WITH_TESTS PIKA_WITH_TESTS_UNIT PIKA_WITH_TESTS_REGRESSIONS are set, and then 'gmake test' only runs ```hello_world_test```.

Version: 0.20.0
FreeBSD 13.2

**Comment by msimberg** [comment]
@yurivict are you building the tests before running the tests (with `cmake --build <build-dir> --target tests`/`gmake tests`)? Could you paste the output of the test command (including `gmake VERBOSE=1 test`/`ctest --verbose`)?

**Comment by yurivict** [comment]
Yes, we build tests first.

Please see the complete [log](https://people.freebsd.org/~yuri/pika-0.20.0-tests.log) of:
* configure-and-build project
* configure-and-build tests
* run tests

**Comment by msimberg** [comment]
Thanks @yurivict, I think I see where the confusion is coming from now.

Am I right in assuming you're not actually calling `ctest` or `make test` anywhere (note, not `make test*s*`)? I couldn't see the actual commands in the log, only the output.

The `hello_world` tests you see at the end are "build unit tests", that confusingly get built when you build the tests, not when you "run" the tests. We might consider making the build unit tests actually test targets, but I think that is not the issue at hand here.

So, in summary, are you calling `ctest` or `make test` after building the tests?

**Comment by msimberg** [comment]
Nice, thanks for confirming!
